### PR TITLE
fix(AIChat): Code block style updates

### PR DESCRIPTION
**Pull request by StephenHeaps**
Resolves https://github.com/brave/brave-browser/issues/39694

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Ask Leo to generate some code responses
2. Verify copy button copies the code in the code block
3. Verify line lines are displayed beside each line (aligned to top line when lines wrap)

iPhone light | iPhone dark
--|--
iPad light | iPad dark

### CODE REVIEW

**Review by Brandon-T — APPROVED** [review]

File: ios/brave-ios/Sources/AIChat/Components/Messages/AIChatResponseMessageView.swift [hunk 1]
```diff
@@ -206,6 +196,76 @@ struct AIChatResponseMessageView: View {
...
+            .foregroundStyle(Color(braveSystemName: .textSecondary))
+          }
+        )
+      }
+      .padding(16)
+    }
+
+    private var codeBlockSection: some View {
+      Grid(alignment: .leading, horizontalSpacing: nil, verticalSpacing: nil) {
+        ForEach(Array(block.string.split(separator: "\n").enumerated()), id: \.offset) {
```
Comment (kylehickinson) [inline ios/brave-ios/Sources/AIChat/Components/Messages/AIChatResponseMessageView.swift:251]: offset is not a stable identifier and shouldn't be used

File: ios/brave-ios/Sources/AIChat/Components/Messages/AIChatResponseMessageView.swift [hunk 2]
```diff
@@ -206,6 +196,76 @@ struct AIChatResponseMessageView: View {
...
+          codeBlock.languageHint?.capitalizeFirstLetter ?? Strings.AIChat.leoCodeExampleDefaultTitle
+        )
+        .foregroundStyle(Color(braveSystemName: .textPrimary))
+        Spacer()
+        Button(
+          action: {
+            UIPasteboard.general.setValue(
+              String(block.string.characters),
+              forPasteboardType: "public.plain-text"
+            )
```
Comment (kylehickinson) [inline ios/brave-ios/Sources/AIChat/Components/Messages/AIChatResponseMessageView.swift:234]: Can we not use `UIPasteboard.general.string = block.string`?

Comment (StephenHeaps) [reply ios/brave-ios/Sources/AIChat/Components/Messages/AIChatResponseMessageView.swift:234]: I was matching previous styling in AIChat target, but I will update those too for readability 👍.

Comment (StephenHeaps) [reply ios/brave-ios/Sources/AIChat/Components/Messages/AIChatResponseMessageView.swift]: Updated in [26f2214](https://github.com/brave/brave-core/pull/24606/commits/26f22148aaa1852ca6765c183d745b03dd31e685)

File: ios/brave-ios/Sources/AIChat/Components/Messages/AIChatResponseMessageView.swift (hunk 1 above)
Comment (StephenHeaps) [reply ios/brave-ios/Sources/AIChat/Components/Messages/AIChatResponseMessageView.swift]: Updated to unique identifier in [65264be](https://github.com/brave/brave-core/pull/24606/commits/65264be48db73e843ca2ffd6356fb972d3b12930)

**Review by kylehickinson — APPROVED** [review]

**Pull request merged**